### PR TITLE
Fix win paths in Nuget tasks #29

### PR DESCRIPTION
**Pull request by yahavi**

### CODE REVIEW

File: tasks/ArtifactoryNuget/nugetBuild.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ function exec(cliPath, nugetCommand) {
     } else {
         // Perform push command.
         let targetDeployRepo = tl.getInput("targetDeployRepo");
-        let pathToNupkg = tl.getInput("pathToNupkg");
+        let pathToNupkg = utils.fixWindowsPaths(tl.getPathInput("pathToNupkg"), true, false);
```
Comment (dnperfors) [inline tasks/ArtifactoryNuget/nugetBuild.js]: I am rather new to this code base, but doesn't the `utils.fixWindowsPaths` function only need one argument?

Comment (yahavi) [reply tasks/ArtifactoryNuget/nugetBuild.js]: Thank you, @dnperfors. You are right!

File: tasks/ArtifactoryNuget/task.json [hunk 2]
```diff
@@ -89,16 +89,16 @@
...
             "required": "true",
             "visibleRule": "command = restore"
         },
         {
             "name": "pathToNupkg",
             "type": "filePath",
             "label": "Path to NuGet package(s) to publish",
-            "defaultValue": "**/*.nupkg",
-            "helpMarkDown": "The path to nupkg file(s) to be uploaded.",
+            "defaultValue": "*.nupkg",
```
Comment (eyalbe4) [inline tasks/ArtifactoryNuget/task.json]: I suggest having the same default value as the default Nuget task, which is:
$(Build.ArtifactStagingDirectory)/**/*.nupkg;!$(Build.ArtifactStagingDirectory)/**/*.symbols.nupkg
We should probably check that we're able to cope with this expression.

File: tasks/ArtifactoryNuget/task.json [hunk 3]
```diff
@@ -89,16 +89,16 @@
...
+            "helpMarkDown": "The path to nupkg file(s) to be uploaded. Accepts wildcards.",
```
Comment (eyalbe4) [inline tasks/ArtifactoryNuget/task.json]: I suggest we modify the *helpMarkDown* message to be the same as the default Nuget task, which is: 
**The pattern to match or path to nupkg files to be uploaded. Multiple patterns can be separated by a semicolon.**

File: tasks/ArtifactoryNuget/nugetBuild.js [hunk 4]
```diff
@@ -136,7 +136,7 @@ function addNugetArgsToCommands() {
         nugetArguments = utils.cliJoin(nugetArguments, "-NoCache");
     }
 
-    let packagesDirectory = tl.getInput("packagesDirectory");
+    let packagesDirectory = utils.fixWindowsPaths(tl.getPathInput("packagesDirectory", false, false));
```
[the remark this answers is not in the document]
Comment (yahavi) [reply tasks/ArtifactoryNuget/nugetBuild.js]: But I changed fixWindowsPaths to:
```
function fixWindowsPaths(string) {
    return isWindows() ? string.replace(/([^\\])\\(?!\\)/g, '$1\\\\') : string;
}
```

File: tasks/ArtifactoryNuget/nugetBuild.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (yahavi) [reply tasks/ArtifactoryNuget/nugetBuild.js]: function fixWindowsPaths(string) {
    return isWindows() ? string.replace(/([^\\])\\(?!\\)/g, '$1\\\\') : string;
}

File: tasks/ArtifactoryNuget/task.json [hunk 5]
```diff
@@ -89,16 +89,16 @@
             "type": "filePath",
             "label": "Path to the solution (The root of working directory by default)",
             "defaultValue": "**/*.sln",
-            "helpMarkDown": "The path to the solution (*.sln).",
+            "helpMarkDown": "The path to the solution (*.sln). Accepts Ant-style patterns.",
```
[the remark this answers is not in the document]
Comment (yahavi) [reply tasks/ArtifactoryNuget/task.json]: But the user need to know that this is an Ant style pattern.

**Pull request merged**